### ISSUE TITLE
Context-Menu COPY is not working in vscode webview, BUT ctrl+c is working fine

### ISSUE THREAD

**Issue by soumasar010322** [microsoft/vscode#141080]
Issue Type: <b>Bug</b>

Use Case 1:
1. Install an extension that uses a Webview (Oracle Developer Tool for VS Code)
2. Connect to a Database
3. Run any SELECT query
4. In Result Tab, we can see the SQL Query that we just ran + Query Result
5. Select the SQL Query Text >> Right Click >> COPY
6. Paste in any text editor

Expected Result: Text is getting copied to text editor
Actual Result: Text is not getting copied to text editor

Use Case 2:
1. Install an extension that uses a Webview (Oracle Developer Tool for VS Code)
2. Connect to a Database
3. Run any SELECT query
4. In Result Tab, we can see the SQL Query that we just ran + Query Result
5. Select the SQL Query Text >> Press Ctrl+C
6. Paste in any text editor

Expected Result: Text is getting copied to text editor
Actual Result: Text is getting copied to text editor

VS Code version: Code 1.63.2 (899d46d82c4c95423fb7e10e68eba52050e30ba3, 2021-12-15T09:40:02.816Z)
OS version: Windows_NT x64 10.0.18363
Restricted Mode: No

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel Xeon E3-12xx v2 (Ivy Bridge, IBRS) (2 x 2693)|
|GPU Status|2d_canvas: unavailable_software<br>gpu_compositing: disabled_software<br>multiple_raster_threads: disabled_off<br>oop_rasterization: disabled_off<br>opengl: disabled_off<br>rasterization: disabled_software<br>skia_renderer: enabled_on<br>video_decode: disabled_software<br>vulkan: disabled_off<br>webgl: unavailable_software<br>webgl2: unavailable_software|
|Load (avg)|undefined|
|Memory (System)|14.65GB (8.33GB free)|
|Process Argv|. --crash-reporter-id 699fa828-d5c2-42a4-95b5-df472e05dbaf|
|Screen Reader|no|
|VM|0%|
</details><details><summary>Extensions (1)</summary>

Extension|Author (truncated)|Version
---|---|---
oracledevtools|Ora|21.3.0

</details><details>
<summary>A/B Experiments</summary>

```
vsliv368cf:30146710
vsreu685:30147344
python383cf:30185419
vspor879:30202332
vspor708:30202333
vspor363:30204092
pythonvspyl392:30422343
pythontb:30283811
pythonptprofiler:30281270
vshan820:30294714
vstes263:30335439
pythondataviewer:30285071
vscod805:30301674
pythonvspyt200:30340761
binariesv615:30325510
bridge0708:30335490
bridge0723:30353136
vsaa593cf:30376535
vsc1dst:30424896
pythonvs932:30410667
vscop804:30404766
vscop453:30404998
vsrem710cf:30416617

```

</details>

**Comment by mjbvz** [comment]
Please put together a minimal example that doesn't rely on an complex extension or connecting to a database 

You can build on the webview sample for this: https://github.com/microsoft/vscode-extension-samples/tree/main/webview-sample

**Comment by soumasar010322** [comment]
@mjbvz , we have already looked into the sample code. In fact, we made the below change in the html returned from **_getHtmlForWebview** method like below:

Original Code:
```
<img src="${catGifPath}" width="300" />
<h1 id="lines-of-code-counter">0</h1>
<script nonce="${nonce}" src="${scriptUri}"></script>
```

Replaced with
```
<h1>This is a Heading</h1>
<p>This is a paragraph.</p>
```

![image](https://user-images.githubusercontent.com/98100790/150469318-950553c6-eabc-4359-bc90-4062a13238c4.png)

After making the change, we tried to copy the text using context menu from the Webview but text is not getting copied.

**Comment by soumasar010322** [comment]
Also, please let us know if there is any way to disable/greyed-out CUT/PASTE item in context menu. We just need the COPY item in the context menu.

**Comment by connor4312** [comment]
Logan and I ran into this in the hex editor today as well.

**Comment by osobeso** [comment]
I'm also seeing this issue with a web view on our extension.
It's really weird, Ctrl+C works perfectly fine, but right-click -> copy does not copy to the clipboard.

I found this, and I think this might be related to the issue:

https://stackoverflow.com/questions/70788421/context-menu-copy-is-not-working-in-vscode-webview-but-ctrlc-is-working-fine

A user replicated this issue while debugging vscode and found out that CTRL+C and right-click->Copy have different implementations.

Does anyone know if there is a workaround for this? Or if there'll be a resolution soon?

Thanks!

**Comment by pomdtr** [comment]
I have the exact inverse problem on Mac os (right click copy is working fine, but cmd+c does not work).

**Comment by palaashatri** [comment]
@mjbvz this issue happens on both Windows as well as on Ubuntu. Text is not copied to clipboard.
It would be a really breeze to use on cases where it's easier to read and copy data from a markdown table for use in an another tool.

For example, I want to copy docker image sizes from a markdown file, while previewing it. Image attached below:
![image](https://user-images.githubusercontent.com/40315170/188867892-bb4d756f-dbd3-45f0-a09e-c86ea7ab0e56.png)

**Comment by ZeWaka** [comment]
Bumping this issue, it's a bit annoying for users when interfacing with a GUI-based extension.

**Comment by CTimmerman** [comment]
> Also, please let us know if there is any way to disable/greyed-out CUT/PASTE item in context menu. We just need the COPY item in the context menu.

https://stackoverflow.com/a/73247563/819417 should fix that and also the copy issue for your extension.

**Comment by Fortinbra** [comment]
My team is running into this same issue with the Thunder Client extension, as well as the extensions market place. We cannot cut/copy/paste from these.

**Comment by evil-pechenka** [comment]
**The context menu copy function is not working**

Does this issue occur when all extensions are disabled?: Yes/No
Yes

VS Code Version: 1.78.2
OS Version: Windows 11 Pro 22H2
Steps to Reproduce:

- I copy using the context menu.
- I paste, but nothing is copied.
- I copy using the keyboard shortcut.
- I paste, and everything is copied.

Please watch the video.

https://github.com/microsoft/vscode/assets/109512933/7cedd1c7-0b0d-47e8-87b3-e82fe9cefec2

**Comment by saritbha** [comment]
I'm using a codegpt extension and im not able to copy. what's the point of codegpt then lol

**Comment by connor4312** [comment]
Should we remove the "copy" context menu action until it's working? It's rather misleading at the moment.

**Comment by StenGruener** [comment]
+1 here on being not able to copy logs from kubernetes addon

**Comment by jwoodard80** [comment]
I am having the issue as well in Thunderclient and markdown extensions. 

Thank you for any work put forward to resolve this.

**Comment by yiliang114** [comment]
ref: https://github.com/microsoft/vscode/issues/205624, Copy Image command in webview is not working neither.

**Comment by yiliang114** [comment]
@mjbvz I found the cause of the problem. When we right-click on the webview, the right-click menu that appears is actually provided by vscode (not webview), and webview mistakenly regarded this situation as losing focus. When the callback function of the copy command is executed, the activeWebview value cannot be obtained, so no operation is done.

When the right-click menu appears, onBlur is executed.
![image](https://github.com/microsoft/vscode/assets/11473889/bae65d67-70c0-40d6-8ade-babc92417322)


The error caused vscode to think that there was no webview in the active state at this time.

https://github.com/microsoft/vscode/blob/00abefa3e27ee1866cf6a6ac825578c1ddafd32c/src/vs/workbench/contrib/webview/browser/webview.contribution.ts#L22~L38

**Comment by yiliang114** [comment]
> @mjbvz I found the cause of the problem. When we right-click on the webview, the right-click menu that appears is actually provided by vscode (not webview), and webview mistakenly regarded this situation as losing focus. When the callback function of the copy command is executed, the activeWebview value cannot be obtained, so no operation is done.我找到了问题的原因。当我们右键单击webview时，出现的右键菜单实际上是由vscode (而不是webview) 提供的，webview错误地将这种情况视为失去焦点。执行copy命令的回调函数时，无法获取activeWebview值，因此不进行任何操作。
> 
> When the right-click menu appears, onBlur is executed.出现右键单击菜单时，将执行onBlur。 ![image](https://private-user-images.githubusercontent.com/11473889/308968316-bae65d67-70c0-40d6-8ade-babc92417322.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDkyMTkxMjEsIm5iZiI6MTcwOTIxODgyMSwicGF0aCI6Ii8xMTQ3Mzg4OS8zMDg5NjgzMTYtYmFlNjVkNjctNzBjMC00MGQ2LThhZGUtYmFiYzkyNDE3MzIyLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAyMjklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMjI5VDE1MDAyMVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTY1MjQ0ZmNlODQxZTY1YmRlZjg2OTI3ODE5MjU3MjNlYThlNzUxZmQ4ZDY3YWRhYjQ5NzkyODFmNzQxMjJiZTEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.o1Baleu2OY2uCAEiV5Ynt26QDZHpLIoy2DGPGbl65L0)
> 
> The error caused vscode to think that there was no webview in the active state at this time.该错误导致vscode认为此时没有处于活动状态的webview。
> 
> https://github.com/microsoft/vscode/blob/00abefa3e27ee1866cf6a6ac825578c1ddafd32c/src/vs/workbench/contrib/webview/browser/webview.contribution.ts#L22~L38

I tried to fix the problem in #206529 : only need to judge whether the right-click menu currently appears when judging whether the webview is focused.
